File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string res="";
        for(char si:s){
            if(isalnum(si)){
                res+=tolower(si);
            }
        }
        int n=res.size();
        return checkPalindrome(0,res,n);
    }

    bool checkPalindrome(int i,string &res , int n){
        if(i>=n/2) return true;
        if(res[i]!=res[n-i-1]) return false;
        return checkPalindrome(i+1, res, n);
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        brr=[]
        for i in s:
            if i.isalnum():
                brr.append(i.lower())
        l,h=0,len(brr)-1
        while l<=h:
            if brr[l]==brr[h]:
                l+=1
                h-=1
            else:
                return False
        return True